presto-hive/src/main/java/com/facebook/presto/hive/statistics/MetastoreHiveStatisticsProvider.java

```diff
@@ -143,7 +146,8 @@ public TableStatistics getTableStatistics(
             return getTableStatistics(columns, columnTypes, partitions, statisticsSample);
         }
         catch (PrestoException e) {
-            if (e.getErrorCode().equals(HIVE_CORRUPTED_COLUMN_STATISTICS.toErrorCode()) && isIgnoreCorruptedStatistics(session)) {
+            if (e.getErrorCode().equals(HIVE_CORRUPTED_COLUMN_STATISTICS.toErrorCode()) && isIgnoreCorruptedStatistics(session)
+                    || e.getErrorCode().equals(HIVE_NULL_COLUMN_NEGATIVE_STATISTICS.toErrorCode()) && isNegativeStatisticCountEnabled(session)) {
```

Comment:
if we meet -1 for nullCounts and the toggle `hive.negative-count-statistics-enabled` is true, then we think the statistics as not present